src/Utility/Hash.php

```diff
@@ -933,12 +933,17 @@ public static function sort(array $data, $path, $dir = 'asc', $type = 'regular')
             $data = array_values($data);
         }
         $sortValues = static::extract($data, $path);
-        $sortCount = count($sortValues);
         $dataCount = count($data);
 
         // Make sortValues match the data length, as some keys could be missing
         // the sorted value path.
-        if ($sortCount < $dataCount) {
+        $missingData = count($sortValues) < $dataCount;
+        if ($missingData && $numeric) {
+            foreach ($data as $key => $value) {
+                // Get the value without the leading '{n}'.
+                $sortValues[$key] = static::get($value, substr($path, 4));
```

Comment:
Might be nitpicking, but couldn't we store the ``substr($path, 4)`` in a var ?
I'm thinking about big arrays to sort. Since we are going through each values, maybe we can micro-optimize this by saving X time the same call ?